### ISSUE TITLE
[harvester] add rke1 cloud provider for harvester

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#35000]
**Feature Request** 
* add and configure `harvester` cloud provider for rke1 clusters

**Additional context**
rke2 is in [but has some bugs](https://github.com/rancher/dashboard/issues/4312), we should also support rke1 cloud provider

**Comment by guangbochen** [comment]
** To Test: **
1. Spin up the latest Rancher server e.g `v2.6.3-rc5` and import the downstream Harvester cluster into the Rancher
2. create a new RKE1 cluster using the Harvester node driver and set the cloud provider to be `Harvester`
![image](https://user-images.githubusercontent.com/4569037/146105292-9b969707-a286-4487-9181-047a854b08c2.png)
3. Once the cluster is up and running, install the Harvester CCM and guest CSI chart manually via clicking the RKE1 cluster -> Apps & Marketplace -> Charts (using the default configuration is enough).
4. Check if the charts are installed successfully via the `Installed Apps` and select All namespaces filter if u can't see those installed apps.
5. Then deploy the app or workload to verify both CSI and CCM features https://docs.harvesterhci.io/v0.3/rancher/cloud-provider
   - verify the `harvester` storage class is the default SC and the workload was able to use it to create a PVC
   -verify users were able to create a Load Balancer type of services and the additional LB configurations can be set via the new `Add-on` tab instead of using original annotations.
![image](https://user-images.githubusercontent.com/4569037/146107886-8ada2b26-f41b-445e-aa04-3487ef36a5f3.png)

**Comment by TachunLin** [comment]
Verified **fixed** on master-3d096946-head (12/15) with rancher `v2.6.3-rc5`. Close this issue.

### Result
1. Can provision RKE1 cluster on harvester with cloud provider to be `Harvester`
![image](https://user-images.githubusercontent.com/29251855/146219488-bb72d3cd-4250-4912-9a2b-b709ad0fb259.png)
2. Can install the Harvester CCM and guest CSI chart manually
![image](https://user-images.githubusercontent.com/29251855/146220089-a261311c-7455-45f6-ac50-3ec9828ce034.png)
3. Can deploy cloud provider and csi driver
4. Can deploy wordpress app with add-on config set health check port
![image](https://user-images.githubusercontent.com/29251855/146340895-56b555b7-fa13-44b8-b65d-9d94bf9829c8.png)
5. Have harvester as default storage class 
![image](https://user-images.githubusercontent.com/29251855/146341081-83d24fe5-acb5-4648-ab92-43498fe62757.png)
6. Deployment apps can create PVC accordingly
![image](https://user-images.githubusercontent.com/29251855/146341233-1d725ec4-a62c-40e6-a759-39317cca9480.png)
7. Additional LB configurations can be set via the new `Add-on` tab in create load balancer page 
![image](https://user-images.githubusercontent.com/29251855/146341573-369084dc-bb3d-44ed-b58b-f5f1d9e78f24.png)

### Test Information
* Test Environment: 1 node harvester on local kvm machine
* Harvester version: master-3d096946-head (12/15)
* Rancher version: v2.6.3-rc5

### Environment Preparation
1. Docker install rancher v2.6.3-rc5  
2. Create one node harvester with enough resource

### Verify Steps

1. Environment preparation as above steps
2. Import harvester to rancher from harvester settings
3. Create cloud credential
4. Create RKE1 node template 
![image](https://user-images.githubusercontent.com/29251855/146299688-3875c18f-61d6-48e6-a15e-250d59c177ba.png)
5. Provision a RKE1 cluster, check the `Harvester` as cloud provider 
![image](https://user-images.githubusercontent.com/29251855/146342214-568bf017-e0e2-4b3a-9f38-894eff77d439.png)
6. Access RKE1 cluster 
7. Open charts in Apps & Market page 
8. Install harvester cloud provider and CSI driver 
9. Make sure cloud provider installed complete
```
NAME: harvester-cloud-provider
LAST DEPLOYED: Thu Dec 16 03:57:26 2021
NAMESPACE: kube-system
STATUS: deployed
REVISION: 1
TEST SUITE: None
---------------------------------------------------------------------
SUCCESS: helm install --namespace=kube-system --timeout=10m0s --values=/home/shell/helm/values-harvester-cloud-provider-100.0.0-up0.1.7.yaml --version=100.0.0+up0.1.7 --wait=true harvester-cloud-provider /home/shell/helm/harvester-cloud-provider-100.0.0-up0.1.7.tgz
```
10. Make sure CSI driver installed complete
```
NAME: harvester-csi-driver
LAST DEPLOYED: Thu Dec 16 03:59:54 2021
NAMESPACE: kube-system
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Successfully deployed Harvester CSI driver to the kube-system namespace.
---------------------------------------------------------------------
SUCCESS: helm install --namespace=kube-system --timeout=10m0s --values=/home/shell/helm/values-harvester-csi-driver-100.0.0-up0.1.8.yaml --version=100.0.0+up0.1.8 --wait=true harvester-csi-driver /home/shell/helm/harvester-csi-driver-100.0.0-up0.1.8.tgz
```
12. Switch to All-namespace and check installed apps page
13. create a new repo and set url to https://charts.bitnami.com/bitnami
![image](https://user-images.githubusercontent.com/29251855/146343516-ff992ee5-9105-42f8-bfa9-42f84a137015.png)

14. Find and deploy `wordpress` in chart 
![image](https://user-images.githubusercontent.com/29251855/146343376-97df8ead-a88e-4493-b78a-b975cc5e21a0.png)

15. Open Global settings in Rancher dashboard 
16. Change the UI-offline preferred to `Remote` 
![image](https://user-images.githubusercontent.com/29251855/146343923-f4051c3a-56b0-4c41-a40e-c664075ed7c0.png)
17. Click ctrl+R to refresh page
18. Access RKE1 cluster 
18. Open `Services` in service discovery
19. Edit the config of wordpress load balancer service
20. Open the Add-on config in load balancer page and save
![image](https://user-images.githubusercontent.com/29251855/146344351-5ea354b7-2931-4b9f-8fe1-40039f008070.png)
21. Check wordpress deployment
22. Check Storage Class page
23. Check Persistent Volume in storage
24. Check the installed apps have wordpress, cloud provider and csi driver deployed
